ecosystem/indexer-grpc/indexer-grpc-manager/src/data_manager.rs

```diff
@@ -0,0 +1,311 @@
+        update_file_store_version: bool,
+    ) -> Vec<Transaction> {
+        if !update_file_store_version {
+            trace!(
+            "Requesting version {start_version} from cache, update_file_store_version = {update_file_store_version}.",
+        );
+            trace!(
+                "Current data range in cache: [{}, {}).",
+                self.start_version,
+                self.start_version + self.transactions.len() as u64
+            );
+        }
+        if start_version < self.start_version {
+            return vec![];
+        }
+
+        let mut transactions = vec![];
+        let mut size_bytes = 0;
+        for transaction in self
+            .transactions
+            .iter()
+            .skip((start_version - self.start_version) as usize)
+        {
+            size_bytes += transaction.encoded_len();
+            transactions.push(transaction.clone());
+            if size_bytes > max_size_bytes {
+                // Note: We choose to not pop the last transaction here, so the size could be
+                // slightly larger than the `max_size_bytes`. This is fine.
+                break;
+            }
+        }
+        if update_file_store_version {
+            self.file_store_version
+                .fetch_add(transactions.len() as u64, Ordering::SeqCst);
+        } else {
+            trace!(
+                "Returned {} transactions from Cache, total {size_bytes} bytes.",
+                transactions.len()
+            );
+        }
+        transactions
+    }
+}
+
+pub(crate) struct DataManager {
+    // TODO(grao): Putting a big lock for now, if necessary we can explore some solution with less
+    // locking / lock-free.
+    cache: RwLock<Cache>,
+    file_store_reader: FileStoreReader,
+    metadata_manager: Arc<MetadataManager>,
+}
+
+impl DataManager {
+    pub(crate) async fn new(
+        chain_id: u64,
+        file_store_config: IndexerGrpcFileStoreConfig,
+        cache_config: CacheConfig,
+        metadata_manager: Arc<MetadataManager>,
+    ) -> Self {
+        let file_store = file_store_config.create_filestore().await;
+        let file_store_reader = FileStoreReader::new(chain_id, file_store).await;
+        let file_store_version = file_store_reader.get_latest_version().await.unwrap();
+        Self {
+            cache: RwLock::new(Cache::new(cache_config, file_store_version)),
+            file_store_reader,
+            metadata_manager,
+        }
+    }
+
+    pub(crate) async fn start(&self) {
+        info!("Starting DataManager loop.");
+
+        'out: loop {
+            let mut fullnode_client = self.metadata_manager.get_fullnode_for_request();
+            let cache = self.cache.read().await;
+            let request = GetTransactionsFromNodeRequest {
+                starting_version: Some(cache.start_version + cache.transactions.len() as u64),
+                transactions_count: Some(100000),
+            };
+            drop(cache);
+
+            debug!(
```

Comment:
if you want to surface these logs up in humio, you may upgrade to info or higher; I was under the impression that only info and above will be collected

Reply:
I choose debug! for this one on purpose. Feels too verbose for info!. I will have enough information on dashboard for version.